### ISSUE TITLE
AccessViolationException in ContactEventsManager.Flush

### ISSUE THREAD

**Issue by JeroMiya** [stride3d/stride#2578]
**Release Type**: Official Release

**Version**: 4.2.0.2282

**Platform(s)**: Windows

**Describe the bug**
I'm not sure what triggers it exactly, but I can reproduce it in my project fairly consistently (see steps below).

**To Reproduce**
1. Download the `bepu-wip` branch of my samples repo: https://github.com/JeroMiya/stride3d-samples/tree/bepu-wip
2. Run the project, select `Level1` to load the scene
3. Wait in the start position for the white sphere to hit you
4. Application crashes with this exception: AccessViolationException

**Expected behavior**
Either a better exception (if it's something wrong with my code) or no exception.

**Screenshots**
![Image](https://github.com/user-attachments/assets/3fdf56aa-3a40-45cf-b7db-bca67210e405)

**Log and callstacks**
See screenshot

**Comment by Eideren** [comment]
Might need need more checks in place higher up the stack when sending events.
Can you delay the `Entity.Scene = null` inside your `void IContactEventHandler.OnStartedTouching` to the next update to see if that is the issue ?

**Comment by JeroMiya** [comment]
> Might need need more checks in place higher up the stack when sending events. Can you delay the `Entity.Scene = null` inside your `void IContactEventHandler.OnStartedTouching` to the next update to see if that is the issue ?

Yep, after delaying the `Entity.Scene = null` to the next `Update`, it no longer crashes.

**Comment by Eideren** [comment]
Cool, thanks for checking, I can't fix this one until after the holidays though. If someone is interested it's in https://github.com/stride3d/stride/blob/master/sources/engine/Stride.BepuPhysics/Stride.BepuPhysics/Definitions/Contacts/ContactEventsManager.cs
Have to check that removing the component while processing its events doesn't break bepu's unmanaged array and indexing we use in both `HandleManifoldInner` and `Flush`

**Comment by luca-domenichini** [comment]
I'm having the same issue (and many others with Bepu Physics colliders).
Pinging here @Eideren to check fix status 😊

**Comment by Eideren** [comment]
Oh, I completely forgot this issue existed, should be fixed in master through #2603
Feel free to open again if it isn't !

**Comment by luca-domenichini** [comment]
@Eideren could you please point me out on how to use master sources instead of nupkgs?

If I build with "dotnet build Stride.sln", I get 4.2.0.1 nupkgs, that are not suitable since current version from nuget feed is 4.2.0.2293.
I cannot even "downgrade" all my projects to this fake/local 4.2.0.1 nupkgs, since I'm using Stride.CommunityToolkit which requires 4.2.0.2293 too.

If I link csproj in my solution, I see Stride.BepuPhysics is a .NET Framework 4 project and there is no way to use in a .net8 proj.

Which is the recommended approach?

**Comment by VaclavElias** [comment]
I will wait what Eideren suggests, but regarding the toolkit, you might need to clone that repo as well and then chnage Stride NuGet packages based on Eideren's suggestion.

**Comment by Eideren** [comment]
Yeah, you should swap your solution to your local 4.2.0.1 and do the same for the toolkit, not sure if you have to do anything special for the toolkit - @VaclavElias likely knows

**Comment by VaclavElias** [comment]
You would need to clone https://github.com/stride3d/stride-community-toolkit and update references to `4.2.0.1`.

**Comment by luca-domenichini** [comment]
Thanks for your support.
I managed to build a `4.2.0.9000` nupkg and use that in my solution by doing the following:

1. git clone Stride repo
2. change Stride version in `SharedAssemblyInfo.cs` to a semver following `4.2.0.2293` --> `4.2.0.9000`
3. run `dotnet build Stride.sln`
4. configure a local filesystem nuget repo at `C:\Users\<user>\AppData\Local\Stride\NugetDev` (this is the folder where nupkgs are copied automatically when running `dotnet build` in this case..)
5. update nuget pkgs in my solution to `4.2.0.9000`

After doing that, I have my csprojs (and Stride.CommunityToolkit too) using my locally built `4.2.0.9000` Stride 🚀.

`AccessViolationException` seems not to happen anymore, but unfortunately, it is very difficult to confirm since my project stucks after running for few seconds, due to #2601.
